src/effects/VST3/VST3Effect.cpp

```diff
@@ -0,0 +1,1138 @@
+/**********************************************************************
+
+  Audacity: A Digital Audio Editor
+
+  @file VST3Effect.cpp
+
+  @author Vitaly Sverchinsky
+
+  @brief Part of Audacity VST3 module
+
+**********************************************************************/
+
+#include "VST3Effect.h"
+
+#include "Base64.h"
+
+#include "widgets/AudacityMessageBox.h"
+
+#include <stdexcept>
+#include <wx/log.h>
+#include <wx/stdpaths.h>
+#include <wx/regex.h>
+#include <pluginterfaces/vst/ivstaudioprocessor.h>
+#include <pluginterfaces/vst/ivsteditcontroller.h>
+#include <pluginterfaces/vst/ivstprocesscontext.h>
+#include <public.sdk/source/vst/hosting/hostclasses.h>
+#include <public.sdk/source/vst/vstpresetfile.h>
+
+#include "internal/ComponentHandler.h"
+#include "internal/ParameterChanges.h"
+#include "internal/PlugFrame.h"
+#include "internal/ConnectionProxy.h"
+
+#include "widgets/NumericTextCtrl.h"
+
+#include "SelectFile.h"
+
+#include "ShuttleGui.h"
+
+#include "VST3Utils.h"
+
+#ifdef __WXMSW__
+#include <ShlObj_core.h>
+#endif
+
+namespace {
+
+Steinberg::Vst::IHostApplication* LocalContext()
+{
+   static Steinberg::Vst::HostApplication localContext;
+   return &localContext;
+}
+
+class PresetsBufferStream : public Steinberg::Vst::BufferStream
+{
+public:
+
+   static PresetsBufferStream* fromString(const wxString& str)
+   {
+      Steinberg::Buffer buffer(str.size() / 4 * 3);
+      auto len = Base64::Decode(str, buffer);
+      wxASSERT(len <= buffer.getSize());
+      buffer.setSize(len);
+
+      auto result = owned(safenew PresetsBufferStream);
+      result->mBuffer.take(buffer);
+      return result.take();
```

Comment:
Returning a "stupid" pointer means I have to check all uses of the function to be convinced there is no leak